### PR TITLE
Publicize: Remove dead code from publicize-options

### PR DESCRIPTION
**Pull request by lamosty**
In this PR, we are removing dead code from `post-editor/editor-sharing/publicize-options`.

Originally, @jsnajdr asked me to do something about the `this.props.site.once()` code – we removed `sites-list` and so the site object is from Redux and doesn't have any methods.

I tried to run the code which would execute that function but couldn't get to it. Opened editor, went to the sidebar, clicked on "Sharing" and still couldn't hit it.

When looking deeper, I found out that the `PublicizeOptions` component is rendered in `editor-sharing/index`. `editor-sharing/index` is rendered only in `editor-sharing/accordion` and only under the condition that the Publicize (Jetpack module) is enabled. "That's okay", you say. However, the code in `publicize-options` is called only if the Publicize (Jetpack module) is not enabled:

```
if ( this.props.site && this.props.site.jetpack && ! this.props.isPublicizeEnabled ) {
```

That makes it dead code to me. Is that correct?

### CODE REVIEW

**Comment by jsnajdr** [comment]
Yes, I can confirm that this has been dead code since at least #7089 (July 2016, by @aduth) that made the `EditorSharingPublicizeOptions` component render `null` if Publicize module is not enabled. I'm not sure if it was intentional, but it seems to not have been missed for a year and a half.

Clicking on the "View Module Settings" would open a `wp-admin` page for activating and deactivating Jetpack modules. I'm not sure if we want to re-enable a feature that sends you to `wp-admin`.

Let's remove it.

File: client/post-editor/editor-sharing/publicize-options.jsx [hunk 1]
```diff
@@ -203,26 +166,6 @@ class EditorSharingPublicizeOptions extends React.Component {
 			);
 		}
 
-		if ( this.props.site && this.props.site.jetpack && ! this.props.isPublicizeEnabled ) {
```
Comment (jsnajdr) [inline client/post-editor/editor-sharing/publicize-options.jsx]: The condition a few lines above that checks for `this.props.site.options.publicize_permanently_disabled` is most likely dead, too. "publicize permanently disabled" implies "publicize disabled".

File: client/post-editor/editor-sharing/publicize-options.jsx [hunk 2]
```diff
@@ -203,26 +166,6 @@ class EditorSharingPublicizeOptions extends React.Component {
...
-			return (
-				<div className="editor-sharing__publicize-disabled">
```
Comment (jsnajdr) [inline client/post-editor/editor-sharing/publicize-options.jsx]: The CSS rules for this class can also be removed. And maybe there's more CSS to remove than this.

File: client/post-editor/editor-sharing/publicize-options.jsx [hunk 3]
```diff
@@ -87,38 +82,6 @@ class EditorSharingPublicizeOptions extends React.Component {
...
 	};
 
-	jetpackModulePopup = () => {
-		let href;
-
-		if ( ! this.props.site || ! this.props.site.jetpack ) {
-			return;
-		}
-
-		href = paths.jetpackModules( this.props.site, 'publicize' );
```
Comment (jsnajdr) [inline client/post-editor/editor-sharing/publicize-options.jsx]: This removes the only usage of `paths.jetpackModules`. You can remove it from `lib/paths`.

File: client/post-editor/editor-sharing/publicize-options.jsx (hunk 1 above)
Comment (lamosty) [reply client/post-editor/editor-sharing/publicize-options.jsx]: Yeah, just tested disabling Publicize on a JP site and that message is not shown. 👍

**Pull request merged**